app/middleware/service-selection.ts

```diff
@@ -1,7 +1,41 @@
 import { NextFunction, Response } from 'express';
 import { ServiceConfig } from '../models/services/base-service';
 import HarmonyRequest from '../models/harmony-request';
-import { chooseServiceConfig } from '../models/services';
+import { chooseServiceConfig, getServiceConfigs } from '../models/services';
+
+/**
+ * Add collections to service configs
+ * @param req - The client request
+ * @returns An array of service configurations
+ */
+function addCollectionsToServicesByAssociation(req: HarmonyRequest): ServiceConfig<unknown>[] {
+  const configs = getServiceConfigs();
+  const collections = req.collections || [];
+  // for every collection in the request...
+  for (const coll of collections) {
+    const services = coll.associations?.services || [];
+    // for every service associated with the current collection...
+    for (const serviceId of services) {
+      for (const config of configs) {
+        // if the service config contains a 'umm_s' entry that includes the current service id
+        if (config.umm_s?.includes(serviceId)) {
+          if (config.collections) {
```

Comment:
This method gets about 8 levels deep in nesting.  I'm wondering if there are ways to simplify for clarity.  It could get a little easier if we ensured config.collections was an array (possibly empty) either externally or at the start of the block with an `if (!config.collections) { config.collections = []; }` and then another clause that's `if (config.umm_s?.includes(serviceId) && !config.collections.includes(coll.id))` or just dropping the check for duplication entirely?